cpp/CMakeLists.txt

```diff
@@ -145,11 +135,52 @@ rapids_export(
   DOCUMENTATION doc_string
 )
 
+set(build_code_string
+    [=[
+
+# Find cuFile and determine which features are supported
+find_package(cuFile)
+if(NOT cuFile_FOUND)
+  message(WARNING "KvikIO: Building KvikIO without cuFile")
+else()
+  file(READ "${cuFile_INCLUDE_DIRS}/cufile.h" CUFILE_H_STR)
```

Comment:
This doesn't seem necessary. Instead you can propagate this information from the current CMake execution into the final code string.

Something like:
```cmake

set(code_string [=[
set(cuFile_BATCH_API_FOUND @cuFile_BATCH_API_FOUND@)
set(cuFile_STREAM_API_FOUND @cuFile_STREAM_API_FOUND@)

# Enable supported cuFile features in KvikIO
if(cuFile_FOUND)
  target_link_libraries(kvikio::kvikio INTERFACE cufile::cuFile_interface)
  target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_CUFILE_FOUND)
  if(cuFile_BATCH_API_FOUND)
    target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_CUFILE_BATCH_API_FOUND)
  endif()
  if(cuFile_STREAM_API_FOUND)
    target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_CUFILE_STREAM_API_FOUND)
  endif()
endif()
]=])
string(CONFIGURE "${code_string}" code_string @ONLY) 

rapids_export(
  INSTALL kvikio
  EXPORT_SET kvikio-exports
  GLOBAL_TARGETS kvikio
  NAMESPACE kvikio::
  DOCUMENTATION doc_string
  FINAL_CODE_BLOCK code_string
)
rapids_export(
  INSTALL kvikio
  EXPORT_SET kvikio-exports
  GLOBAL_TARGETS kvikio
  NAMESPACE kvikio::
  DOCUMENTATION doc_string
  FINAL_CODE_BLOCK code_string
)
```